.github/workflows/tests.yml

```diff
@@ -15,76 +15,101 @@ on:
 
 env:
   TQDM_DISABLE: 1
+  CI_SLACK_CHANNEL: ${{ secrets.CI_PUSH_MAIN_CHANNEL }}
 
 jobs:
-  check_code_quality:
-    runs-on: ubuntu-latest
+  tests:
+    name: Tests
     strategy:
       matrix:
-        python-version: [3.9]
-
+        python-version: ['3.9', '3.10', '3.11', '3.12']
+        os: ['ubuntu-latest', 'windows-latest']
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v4
-        with:
-          fetch-depth: 0
-          submodules: recursive
       - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
-      - uses: pre-commit/action@v3.0.1
+          cache: "pip"
+          cache-dependency-path: |
+              setup.py
+              requirements.txt
```

Comment:
If we find the dependencies are a hassle to maintain, we might want to switch to the `transformers` approach, which uses a look-up table to give fine-grained control: https://github.com/huggingface/transformers/blob/main/setup.py